### ISSUE TITLE
Problem: "could not find implicit value for evidence parameter of type kantan.csv.HeaderDecoder"

### ISSUE THREAD

**Issue by james-mcgivern** [nrinaudo/kantan.csv#107]
Hi

I'm trying to get a very simple setup to work however I'm getting the following error `Error:(27, 55) could not find implicit value for evidence parameter of type kantan.csv.HeaderDecoder[integrationtests.ExportedEvent]
      val reader = dataPath.asCsvReader[ExportedEvent](rfc.withHeader)`    

    import java.time.LocalDateTime
    import kantan.csv._
    import kantan.csv.ops._
    import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpec}
    
    case class ExportedEvent(uuid: String, date: LocalDateTime, timestamp: Long, name: String, systemId: String, sensorId: String, location: String, eventType: String, deviceSubType: String, value: Double, units: String, init: String, partitionId: String, siaCode: String, status: String)
    
    class SaintMonicaFalsePositiveIntegrationSpec extends WordSpec
      with Matchers
      with BeforeAndAfterAll {
    
      "The system" when {
        "is processing data" in {
          val dataPath = getClass.getResource("/integration-tests/saint-monica-20171202-incident/data.csv")
          val reader = dataPath.asCsvReader[ExportedEvent](rfc.withHeader)
          reader.foreach(println _)
        }
      }
    }

Please could you advise on what I'm missing, here's the dependencies in my Maven POM:

        <dependency>
            <groupId>com.nrinaudo</groupId>
            <artifactId>kantan.csv_2.11</artifactId>
            <version>0.3.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>com.nrinaudo</groupId>
            <artifactId>kantan.csv-java8_2.11</artifactId>
            <version>0.3.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>com.nrinaudo</groupId>
            <artifactId>kantan.csv-cats_2.11</artifactId>
            <version>0.3.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>com.nrinaudo</groupId>
            <artifactId>kantan.csv-generic_2.11</artifactId>
            <version>0.3.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>com.nrinaudo</groupId>
            <artifactId>kantan.csv-refined_2.11</artifactId>
            <version>0.3.0</version>
            <scope>test</scope>
        </dependency>

Thanks
James

**Comment by james-mcgivern** [comment]
Ok so I figured out I was missing a `HeaderEncoder` so I fixed that with 

          implicit val headerDecoder = HeaderDecoder.decoder("uuid", "date", "timestamp", "name", "systemId", "shepherdId", "location", "type", "deviceSubType", "value", "units", "init", "partitionId", "siaCode", "status")(ExportedEvent.apply _)

However now I'm getting `Error:(27, 219) could not find implicit value for evidence parameter of type kantan.csv.CellDecoder[java.time.LocalDateTime]
      implicit val headerDecoder = HeaderDecoder.decoder("uuid", "date", "timestamp", "name", "systemId", "shepherdId", "location", "type", "deviceSubType", "value", "units", "init", "partitionId", "siaCode", "status")(ExportedEvent.apply _)`

I think I can fix this by defining an implicit `CellDecoder` but I thought having the `kantan.csv-java8` artifact on the classpath would mean I didn't have to do this

**Comment by nrinaudo** [comment]
No, that's not how things work in Scala - we usually try to avoid magic like _the code behaviour changes depending on whether a JAR is in the CLASSPATH_.

In your case, it's good that you have the `java8` module on the CLASSPATH. But if you look at [the documentation](http://nrinaudo.github.io/kantan.csv/tut/java8.html), you also need to explicitly impor the corresponding values:

```scala
import kantan.csv.java8._
```

**Comment by james-mcgivern** [comment]
Thank you! I'm just getting to grips with serious Scala but I should have read the docs closer.

**Comment by nrinaudo** [comment]
No worries. I didn't mean that you should have read the docs, but that they were there in case you still struggled with my answer.

Did that deal with your issue? If so, would you mind closing this? If not, do let me know what's wrong and I'll try to help you through it.

**Comment by james-mcgivern** [comment]
We should all RTM :)
